### ISSUE TITLE
OS X: Beachball crash for [7f01f7f] (SVN r754) opening v3.0.9 savefile with randarts

### ISSUE THREAD

**Issue by angband-import** [angband/angband#1025]
**Reported by roustk@alumni.caltech.edu on 9 Mar 2008 05:32 UTC**
Simple sequence which reliably crashes [7f01f7f] (SVN r754) under MacOS 10.5.2:

1. Open Angband v3.0.9 and create and save a new character (default options plus randarts).

2. Open Angband v754 and open the saved character.

3. Angband crashes in one of two modes:
  a. Beachball lockup (requiring force quit).
  b. Angband error popup saying "Out of memory!" while Angband window says
```
  Loading a 3.0.10 savefile...
  Incompatible random artifacts version!
```

Failing to open the file is correct behavior; crashing is not.

It appears that the problem lies in load.c in function old_load().  At line 2331, when loading fails, we return -1 into err and:
```
  if (err) what = "Cannot parse savefile";
```

Unfortunately, at line 2280, we defined:
```
  cptr what = "generic";
```

Why would this be a problem?  cptr is a const char * -- we shouldn't assign things to what, and we *really* shouldn't assign a string longer than 7 characters.  This causes the memory error, and probably causes the beachball.

Kevin

**Comment by angband-import** [comment]
**Modified by takkaria on 17 May 2008 01:27 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 28 Jun 2008 10:15 UTC**
should be fixed in [fa6992f] (SVN r966).
